transaction/src/transaction/builder.rs

```diff
@@ -193,6 +197,15 @@ impl Builder {
         self
     }
 
+    pub fn add_validator_definition(
+        &mut self,
+        validator: Validator,
+        signing_key: SigningKey<SpendAuth>,
+    ) -> &mut Self {
+        self.validator_definitions.push((validator, signing_key));
+        self
+    }
+
```

Comment:
It looks like this is doing deferred signing, but I think that it might be easier and cleaner to do immediate signing, or even to just take a `ValidatorDefinition` and have the caller do the signing.

Unlike the case of a spend, where it's super important that the spend happen in the context of *a particular transaction* (i.e., the spend authorization should be bound also to the outputs, so you can't take my spends and put them in your transaction), the validator definitions are essentially context-free: they're signed by the validator's identity key, but that's it, and the purpose of the transaction is just to get the self-authenticating `ValidatorDefinition` to the chain.

If we had the code take a signed `ValidatorDefinition`, it'd also be possible to support offline setups mostly-free: the offline machine with the identity key signs the validator definition, it gets copied over to the machine with the hot wallet, then uploaded with no further intervention, etc.

Reply:
Yeah, the signature in the `ValidatorDefinition` has to only be a signature over the `Validator` itself, because the point is that the `ValidatorDefinition` is supposed to be a self-authenticating definition of some validator configuration, and we should be able to pass it around the program without needing to hold on to other context.  So we don't want to have the signature sign over other data (e.g., the rest of the transaction that broadcast it), because then we wouldn't be able to verify it as a standalone item.

But in that case, there's no need for deferred signing, since the `ValidatorDefinition` isn't bound to any of the other transaction data.